From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: doc: split buffer.rst out of api-summary.rst
Date: Tue, 16 Apr 2024 04:17:52 +0100

Buffer heads are no longer a generic filesystem API but an optional
filesystem support library.  Make the documentation structure reflect
that, and include the fine documentation kept in buffer_head.h.  We could
give a better overview of what buffer heads are all about, but my
enthusiasm for documenting it is limited.

Link: https://lkml.kernel.org/r/20240416031754.4076917-9-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Tested-by: Randy Dunlap <rdunlap@infradead.org>
Cc: Pankaj Raghav <p.raghav@samsung.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 Documentation/filesystems/api-summary.rst |    3 ---
 Documentation/filesystems/index.rst       |    1 +
 2 files changed, 1 insertion(+), 3 deletions(-)

--- a/Documentation/filesystems/api-summary.rst~doc-split-bufferrst-out-of-api-summaryrst
+++ a/Documentation/filesystems/api-summary.rst
@@ -56,9 +56,6 @@ Other Functions
 .. kernel-doc:: fs/namei.c
    :export:
 
-.. kernel-doc:: fs/buffer.c
-   :export:
-
 .. kernel-doc:: block/bio.c
    :export:
 
--- a/Documentation/filesystems/index.rst~doc-split-bufferrst-out-of-api-summaryrst
+++ a/Documentation/filesystems/index.rst
@@ -50,6 +50,7 @@ filesystem implementations.
 .. toctree::
    :maxdepth: 2
 
+   buffer
    journalling
    fscrypt
    fsverity
_
